File: squares-of-a-sorted-array/squares-of-a-sorted-array.py
class Solution(object):
    def sortedSquares(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        ans = [None] * len(nums)
        left, right = 0, len(nums)-1
        
        for i in range(len(nums)-1, -1, -1):
            if abs(nums[left]) > abs(nums[right]):
                ans[i] = nums[left]**2
                left += 1
            else:
                ans[i] = nums[right]**2
                right -= 1
        return ans
                

File: squares-of-a-sorted-array/squares-of-a-sorted-array.rb
# @param {Integer[]} nums
# @return {Integer[]}
def sorted_squares(nums)
    ans = [0]*nums.length
    
    left, right, i = 0, nums.length-1, nums.length-1
    
    while left <= right
       if nums[left]**2 > nums[right]**2
           ans[i] = nums[left]**2
           left += 1
       else
           ans[i] = nums[right]**2
           right -= 1
       end
        i -= 1
    end
    ans
end